ydb/core/tx/columnshard/engines/changes/general_compaction.cpp

```diff
@@ -275,4 +275,30 @@ std::shared_ptr<TGeneralCompactColumnEngineChanges::IMemoryPredictor> TGeneralCo
     }
 }
 
+ui64 TGeneralCompactColumnEngineChanges::TMemoryPredictorChunkedPolicy::AddPortion(const TPortionInfo& portionInfo) {
+    SumMemoryFix += portionInfo.GetRecordsCount() * (2 * sizeof(ui64) + sizeof(ui32) + sizeof(ui16));
+    ++PortionsCount;
+    THashMap<ui32, ui64> maxChunkSizeByColumn;
+    for (auto&& i : portionInfo.GetRecords()) {
+        SumMemoryFix += i.BlobRange.Size;
+        auto it = maxChunkSizeByColumn.find(i.GetColumnId());
+        if (it == maxChunkSizeByColumn.end()) {
+            it = maxChunkSizeByColumn.emplace(i.GetColumnId(), i.GetMeta().GetRawBytesVerified()).first;
```

Comment:
why do we assign a new value to the it variable? we will overwrite it at the next iteration.